pkg/virt-launcher/virtwrap/live-migration-source.go

```diff
@@ -648,6 +705,37 @@ func (m *migrationMonitor) startMonitor() {
 	}
 }
 
+func (l *LibvirtDomainManager) getJobStatsWithRetry(dom cli.VirDomain) (*libvirt.DomainJobInfo, error) {
+	connectionInterval := 10 * time.Second
+	connectionTimeout := 60 * time.Second
+	var stats *libvirt.DomainJobInfo
+
+	err := utilwait.PollImmediate(connectionInterval, connectionTimeout, func() (done bool, err error) {
+		stats, err = dom.GetJobStats(0)
```

Comment:
Maybe we should be using the `DOMAIN_JOB_STATS_KEEP_COMPLETED` flag here `https://github.com/libvirt/libvirt-go/blob/fa4482e6bba3ecfdac0cb421ade77da135492360/domain.go#L714` 

so we always get the most recent stats, even for completed jobs.

Then before we execute a new migration, maybe we could force a call to GetJobStats() wiht the `DOMAIN_JOB_STATS_COMPLETED` flag, which appears to allow a single read of the last jobs compelted results, then the job resutls are cleared. Maybe it works kind of like "burn after reading"?

So, before migration, clear previous cached job results `DOMAIN_JOB_STATS_COMPLETED` . During migration always view both active and completed  job results `DOMAIN_JOB_STATS_KEEP_COMPLETED`

would that make sense?

Reply:
it doesn't work like this.  We'll still get DOMAIN_JOB_NONE from getjobstats after the job is completed, we'll get some status related to the job, but the job type is still none.